### ISSUE TITLE
Report portal auto analysis doesn't work with serenity-rp-integration for Kotlin based project 

### ISSUE THREAD

**Issue by BushnevYuri** [Invictum/serenity-reportportal-integration#65]
Hello,

I was trying to use the report portal and it worked quite well with JUnit integration. However, once I try to use it with serenity integration, I have some troubles in making an auto-analysis. It just doesn't work without any errors in logs. 

This is my report portal configuration:
<img width="1182" alt="Screenshot 2019-03-13 at 3 21 16" src="https://user-images.githubusercontent.com/4463392/54246819-2656f500-453f-11e9-9dd3-4cb1a597c705.png">

If I manually analysis the same error, again and again, it is not analysed automation into next launches:
<img width="1208" alt="Screenshot 2019-03-13 at 3 22 45" src="https://user-images.githubusercontent.com/4463392/54246871-4d152b80-453f-11e9-847c-8017d9392e4e.png">

I tried to cut everything from my framework and left only one extremely basic example:
https://drive.google.com/file/d/1bSAounA48-wtR53QvpnRwq-vJE6UfhGw/view?usp=sharing

To run related tests:
cd /tests/backend-tests
mvn clean verify -Dtags='type:HealthCheck'

I would very much appreciate if you can point out if it is serenity integration plugin problem. Once I refactored it for using just JUnit, it was working fine. 

Worth to mention, that I use Kotlin. However, not sure if it plays any role here...

**Comment by Invictum** [comment]
@BushnevYuri

Integration only uploads Serenity results to Report Portal service and doesn’t control auto analysis feature itself.

As I remember RP start to automatically tag results after 24 hours from the first manual tagging fact (with clean indexes). A lot of attributes are taken into account during auto analysis - name, tags and parameters, so make sure you run the same tests.

**Comment by BushnevYuri** [comment]
That's interesting because once I do exactly the same test but change it to just JUnit without serenity and use JUnit dependency for Report Portal then the analysis is doing well and it is analyzed right away. 
Tried also to clean up indexes and from scratch seeing the same symptom (working with pure JUnit but doesn't work for serenity via integration).

Will try again a bit later to check if I missed anything.

**Comment by Invictum** [comment]
@BushnevYuri 
Any updates on this issue?

**Comment by BushnevYuri** [comment]
@Invictum unfortunately, didn't have time to look into at all... Last thing which I could see is that for some reason, errors from the report portal were not propagated to the ReportPortal. 
For example, I see an error in serenity report but this error is not presented in the ReportPortal report. In the same time, the status of the test case is failed (but just without error specified).

As ReportPortal using errors for analysis then it is expected that it doesn't work. As I'm using RestAssured, there is some chance that it was related to https://github.com/Invictum/serenity-reportportal-integration/issues/64. 

Also, I just remembered that I use AssertJ for assertions. So need to check if errors coming from these assertions propagated to ReportPortal via the plugin. 

In any case, I'll back as soon as have a chance to recheck. Thanks  for checking on this.

**Comment by DzmitryHumianiuk** [comment]
you have to have ERROR type of the logs (colored in RED) on UI.
Since RP uses only this part of logs for auto-analysis.

**Comment by BushnevYuri** [comment]
right. So it depends on the internal implementation of Serenity and why some of the asserions are not being sent as ERROR. 
In any case, we will try to integrate this once again in a week and will update the ticket accordingly right after that.

**Comment by Invictum** [comment]
@BushnevYuri 

Looks the root cause was in `TREE_OPTIMIZED` handler type.
```
    @Test
    fun `verify fe-edit service`() {
        val configuration = ReportIntegrationConfig.get()
        configuration.useHandler(HandlerType.TREE).usePreset(LogsPreset.TREE_OPTIMIZED)
        val response = healthCheckSteps.ping_auth_service()
        assertThat(response.statusCode).isEqualTo(200)
    }
```

It was removed starting from `1.4.0`. With default configuration all works as expected (at least from my perspective).

**Comment by BushnevYuri** [comment]
Thank you @Invictum !!!
Yes, works great after the update. 

Still couldn't run it due to new issue (looks like happening when we have tests running in parallel within one suite). But we will share that in a separate ticket. Thank you very much for your help!
